mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,459 @@
+        mlflow_model.signature = signature
+
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    # update flavor info to mlflow_model
+    mlflow_model.add_flavor(
+        FLAVOR_NAME,
+        code=code_dir_subpath,
+        version=promptflow.__version__,
+        entry=f"{_MODEL_FLOW_DIRECTORY}/{DAG_FILE_NAME}",
+        **flow_env,
+    )
+
+    # append loader_module, data and env data to mlflow_model
+    pyfunc.add_to_model(
+        mlflow_model,
+        loader_module="mlflow.promptflow",
+        conda_env=_CONDA_ENV_FILE_NAME,
+        python_env=_PYTHON_ENV_FILE_NAME,
+        code=code_dir_subpath,
+        model_config=model_config,
+    )
+
+    # save mlflow_model to path/MLmodel
+    mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
+
+    if conda_env is None:
+        if pip_requirements is None:
+            default_reqs = get_default_pip_requirements()
+            inferred_reqs = mlflow.models.infer_pip_requirements(
+                path, FLAVOR_NAME, fallback=default_reqs
+            )
+            default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+        else:
+            default_reqs = None
+        conda_env, pip_requirements, pip_constraints = _process_pip_requirements(
+            default_reqs,
+            pip_requirements,
+            extra_pip_requirements,
+        )
+    else:
+        conda_env, pip_requirements, pip_constraints = _process_conda_env(conda_env)
+
+    with open(os.path.join(path, _CONDA_ENV_FILE_NAME), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    # Save `constraints.txt` if necessary
+    if pip_constraints:
+        write_to(os.path.join(path, _CONSTRAINTS_FILE_NAME), "\n".join(pip_constraints))
+
+    # Save `requirements.txt`
+    write_to(os.path.join(path, _REQUIREMENTS_FILE_NAME), "\n".join(pip_requirements))
+
+    _PythonEnv.current().to_yaml(os.path.join(path, _PYTHON_ENV_FILE_NAME))
+
+
+def _resolve_env_from_flow(flow_dag_path):
+    with open(flow_dag_path) as f:
+        flow_dict = yaml.safe_load(f)
+    environment = flow_dict.get("environment", {})
+    if _FLOW_ENV_REQUIREMENTS in environment:
+        # Append entry path to requirements
+        environment[
+            _FLOW_ENV_REQUIREMENTS
+        ] = f"{_MODEL_FLOW_DIRECTORY}/{environment[_FLOW_ENV_REQUIREMENTS]}"
+    return environment
+
+
+class _PromptflowModelWrapper:
+    def __init__(self, model, model_config: Optional[Dict[str, Any]] = None):
+        from promptflow._sdk._mlflow import FlowInvoker
+
+        self.model = model
+        # TODO: Improve this if we have more configs afterwards
+        model_config = model_config or {}
+        connection_provider = model_config.get("connection.provider", "local")
+        self.model_invoker = FlowInvoker(self.model, connection_provider=connection_provider)
```

Comment:
seems like we missed the connection name override support here. Flow invoker can accept a  connection name override dict which can be also configured via the model_config

Reply:
Added. Minor version updated.